services/venice-controller/src/main/java/com/linkedin/venice/controller/StoreBackupVersionCleanupService.java

```diff
@@ -91,6 +125,83 @@ protected static boolean whetherStoreReadyToBeCleanup(Store store, long defaultB
     return store.getLatestVersionPromoteToCurrentTimestamp() + backupVersionRetentionMs < time.getMilliseconds();
   }
 
+  private boolean validateAllRouterOnCurrentVersion(Store store, String clusterName, int versionToValidate) {
+    List<String> urlList = admin.getHelixVeniceClusterResources(clusterName)
+        .getRoutersClusterManager()
+        .getLiveRouterInstances()
+        .stream()
+        .map(
+            instance -> urlMap
+                .computeIfAbsent(instance, k -> HelixUtils.getInstanceFromHelixInstanceName(k).getUrl(true)))
+        .collect(Collectors.toList());
+
+    for (String routerURL: urlList) {
+      try {
+        HttpGet routerRequest = new HttpGet(routerURL + "/" + TYPE_CURRENT_VERSION + "/" + store.getName());
+        HttpResponse response = storeClient.execute(routerRequest, null).get();
```

Comment:
Let us specify a timeout here to avoid infinite wait?

Reply:
Sure